language/vm/vm-runtime/vm-runtime-types/src/values/values_impl.rs

```diff
@@ -0,0 +1,2223 @@
+    ) -> VMResult<NativeResult> {
+        ensure_len!(ty_args, 1, "type arguments", "destroy empty");
+        ensure_len!(args, 1, "arguments", "destroy empty");
+
+        let cost = native_gas(cost_table, NativeCostIndex::DESTROY_EMPTY, 1);
+        let v = args.pop_front().unwrap().value_as::<Container>()?;
+
+        let is_empty = match (&ty_args[0], &v) {
+            (TypeTag::U8, Container::U8(v)) => v.is_empty(),
+            (TypeTag::U64, Container::U64(v)) => v.is_empty(),
+            (TypeTag::U128, Container::U128(v)) => v.is_empty(),
+            (TypeTag::Bool, Container::Bool(v)) => v.is_empty(),
+
+            (TypeTag::Struct(_), Container::General(v))
+            | (TypeTag::ByteArray, Container::General(v))
+            | (TypeTag::Address, Container::General(v)) => v.is_empty(),
+
+            _ => err_vector_elem_ty_mismatch!(),
+        };
+
+        if is_empty {
+            Ok(NativeResult::ok(cost, vec![]))
+        } else {
+            Ok(NativeResult::err(
+                cost,
+                VMStatus::new(StatusCode::NATIVE_FUNCTION_ERROR)
+                    .with_sub_status(DESTROY_NON_EMPTY_VEC),
+            ))
+        }
+    }
+
+    pub fn native_swap(
+        ty_args: Vec<TypeTag>,
+        mut args: VecDeque<Value>,
+        cost_table: &CostTable,
+    ) -> VMResult<NativeResult> {
+        ensure_len!(ty_args, 1, "type arguments", "swap");
+        ensure_len!(args, 3, "arguments", "swap");
+
+        let cost = native_gas(cost_table, NativeCostIndex::SWAP, 1);
+        let r = pop_arg_front!(args, ContainerRef);
+        let mut v = r.try_borrow_mut()?;
+        let idx1 = pop_arg_front!(args, u64) as usize;
+        let idx2 = pop_arg_front!(args, u64) as usize;
+
+        macro_rules! swap {
+            ($v: ident) => {{
+                if idx1 >= $v.len() || idx2 >= $v.len() {
+                    return Ok(NativeResult::err(
+                        cost,
+                        VMStatus::new(StatusCode::NATIVE_FUNCTION_ERROR)
+                            .with_sub_status(INDEX_OUT_OF_BOUNDS),
+                    ));
+                }
+                $v.swap(idx1, idx2);
+            }};
+        }
+
+        match (&ty_args[0], &mut *v) {
+            (TypeTag::U64, Container::U64(v)) => swap!(v),
+            (TypeTag::Bool, Container::Bool(v)) => swap!(v),
+            (TypeTag::Struct(_), Container::General(v))
+            | (TypeTag::Address, Container::General(v))
+            | (TypeTag::ByteArray, Container::General(v)) => swap!(v),
+            _ => err_vector_elem_ty_mismatch!(),
+        }
+
+        Ok(NativeResult::ok(cost, vec![]))
+    }
+}
+
+/***************************************************************************************
+ *
+ * Gas
+ *
+ *   Abstract memory sizes of the VM values.
+ *
+ **************************************************************************************/
+
+impl Container {
+    fn size(&self) -> VMResult<AbstractMemorySize<GasCarrier>> {
```

Comment:
can you explain what are the reason for size to fail? it used to never fail which is always a good thing and do not understand why it can fail now